src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolBlock2.cs

```diff
@@ -74,6 +74,9 @@ protected MemoryPoolBlock2()
         /// </summary>
         public MemoryPoolBlock2 Next { get; set; }
 
+#if DEBUG
```

Comment:
Maybe we should have a Debug.Assert for when `Slab != null` to ensure that we are always returning the block (in our tests at least).